ee/control/consumers/tokenconsumer/ingest_token_consumer.go

```diff
@@ -0,0 +1,41 @@
+package tokenconsumer
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/kolide/launcher/pkg/agent/types"
+)
+
+const IngestSubsystem = "ingest"
```

Comment:
The functionality of the code all looks pretty good ... now we quibble about naming =). I think this subsystem should just be called `tokens` or `auth_tokens`. Since it's not necessarily related to any type of ingest. We may use it these tokens later to fetch data from somewhere.

If this is only intended to be used for `ObservabiltyIngest` then I think it should be named so

Reply:
discussed here https://github.com/kolide/launcher/pull/1226#discussion_r1230219621